client/src/js/components/bhFindInvoice.js

```diff
@@ -0,0 +1,96 @@
+angular.module('bhima.components')
+  .component('bhFindInvoice', {
+    controller : FindInvoiceComponent,
+    templateUrl : 'modules/templates/bhFindInvoice.tmpl.html',
+    bindings : {
+      patientUuid :       '<', // patient uuid - to restrict search to this patient
+      onSearchComplete :  '&', // bind callback to call when data is available
+      disabled :          '<', // bind disable behavior
+    },
+  });
+
+FindInvoiceComponent.$inject = [
+  'PatientService', 'PatientInvoiceService', 'AppCache', 'NotifyService',
+];
+
+/**
+ * The Find Invoice Component
+ */
+function FindInvoiceComponent(Patients, PatientInvoice, AppCache, Notify) {
+  const vm = this;
+
+  /* @const the enter key keycode */
+  const ENTER_KEY = 13;
+
+  const translate = {
+    name : '',
+  };
+
+  vm.$onInit = function onInit() {
+    vm.disabled = vm.disabled || false;
+  };
+
+  /* Expose functions and variables to the template view */
+  vm.search = search;
+  vm.onKeyPress = onKeyPress;
+  vm.translate = translate;
+
+  /**
+   * @method search
+   */
+  function search(form) {
+    form.$setSubmitted();
+
+    PatientInvoice.findConsumableInvoicePatient(vm.invoiceReference, vm.patientUuid)
+      .then(invoice => {
+        if (!invoice.details) {
+          vm.invoiceFound = false;
+          return;
+        }
+        selectInvoice(invoice);
+      })
+      .catch(Notify.handleError);
+  }
+
+  /**
+   * @method selectInvoice
+   *
+   * @param {object} invoice The invoice object
+   */
+  function selectInvoice(invoice) {
+    vm.invoiceFound = true;
+
+    const elementId = 'search';
+    const searchButton = document.getElementById(elementId);
```

Comment:
Would it be possible to use `$document` instead of `document` here?  It will make it easier to test.